src/toolkits/object_detection/object_detector.cpp

```diff
@@ -959,6 +959,13 @@ std::shared_ptr<MLModelWrapper> object_detector::export_to_coreml(
 std::unique_ptr<data_iterator> object_detector::create_iterator(
     gl_sframe data, std::vector<std::string> class_labels, bool repeat) const
 {
+  // Check annotations type. If it is a single dictionary, put it into a list.
+  const std::string& annotations_column_name =
+      read_state<flex_string>("annotations");
+  if (data.contains_column(annotations_column_name)) {
+    change_annotations_type(data, annotations_column_name);
```

Comment:
Functions that modify their arguments are not as obvious/intuitive to read. Could we move the call to `replace_add_column` to the front, so that this line reads something more like:
```
data[annotations_column_name] = canonicalize_annotations(data[annotations_column_name]);
```
Then it's more immediately obvious what's happening here